plugin/diagnostics.py

```diff
@@ -247,7 +247,7 @@ def end_file(self, file_name: str) -> None:
         self._relevant_file = False
 
     def end(self) -> None:
-        for severity in range(DiagnosticSeverity.Error, DiagnosticSeverity.Hint):
+        for severity in range(DiagnosticSeverity.Error, settings.show_highlights_severity_level + 1):
```

Comment:
First, thanks for fixing this :) 

I noted that you added a new setting `show_highlights_severity_level`.
I would not increase the number of settings.

Can we instead just use the `show_diagnostics_severity_level`?
It makes sense that `show_diagnostics_severity_level` setting
should affect all the diagnostics in the panel,
but that it should also affect all the diagnostics highlights in the document.

Reply:
@predragnikolic @rchl

This should definitely be merged instead of #941 using `settings.show_diagnostics_severity_level`
```suggestion
        for severity in range(DiagnosticSeverity.Error, settings.show_diagnostics_severity_level + 1):
```
TypeScript language server likes to over report diagnostics, obnoxiously drawing squiggly lines all over the code I'm trying to write.

For example, the `unused variable` diagnostic draws a squiggly line under the variable I _just_ wrote.

Reply:
Introducing new settings is something I'd personally rather avoid.
Another option could be to let the diagnostics panel show all severities always and rename existing `show_diagnostics_severity_level` to `show_highlights_severity_level`.

Reply:
I'm proposing to use the existing setting `show_diagnostics_severity_level` so there's no need for an additional setting or renaming.

Reply:
I personally would assume the `highlighting` is a sub-feature of the `diagnostics`.

Reply:
`show_diagnostics_severity_level` currently affects diagnostics shown in the panel and diagnostic highlights. Keeping the existing name and not make it affect diagnostics in the panel would make the meaning of that setting inaccurate. That's why I think it should be renamed if we don't want it to apply to the panel anymore.

Because I guess the whole point of raising this issue is that you want to use different levels for the panel and for the highlights.

Reply:
Then the `show_diagnostics_severity_level` does not correctly match the diagnostics shown in the panel with the document diagnostic highlights.

For example,

## `"show_diagnostics_severity_level": 2,`

<img width="720" alt="Screen Shot 2020-05-21 at 6 02 46 AM" src="https://user-images.githubusercontent.com/1457327/82548561-7a893c00-9b29-11ea-8fc0-837c1d97871d.png">

Notice how the diagnostics panel **only** shows `errors`, yet the document still renders `hint` highlights.

## `"show_diagnostics_severity_level": 4,`

<img width="720" alt="Screen Shot 2020-05-21 at 6 03 37 AM" src="https://user-images.githubusercontent.com/1457327/82548880-0c914480-9b2a-11ea-807d-ca5f63b4df56.png">

Notice how the diagnostics panel shows `errors` and `hints`, and the document correctly renders both `errors` and `hints`.

Reply:
This is not how it works for me.
Make sure that you restart after changing the `show_diagnostics_severity_level` setting for good measure.
Also, are you sure those highlights are from typescript server? Should be clear when hovering them.

Reply:
Yes I'm very familiar with the `Quit Sublime Text && Open Sublime Text` workflow after making _any_ settings.

Yes the `typescript-language-server` is sending `:: <-  ts-lsp textDocument/publishDiagnostics:`
```json
{
	"diagnostics": [
		{
			"source": "typescript",
			"range": {
				"end": {
					"character": 66,
					"line": 10
				},
				"start": {
					"character": 41,
					"line": 10
				}
			},
			"relatedInformation": [
				{
					"location": {
						"uri": "file:///Users/user/Projects/futon-media/node_modules/playwright-core/types/types.d.ts",
						"range": {
							"end": {
								"character": 45,
								"line": 2293
							},
							"start": {
								"character": 26,
								"line": 2293
							}
						}
					},
					"message": "An argument for 'userDataDir' was not provided."
				}
			],
			"severity": 1,
			"message": "Expected 1-2 arguments, but got 0.",
			"code": 2554
		},
		{
			"severity": 4,
			"source": "typescript",
			"range": {
				"end": {
					"character": 27,
					"line": 1
				},
				"start": {
					"character": 0,
					"line": 1
				}
			},
			"code": 6133,
			"message": "'R' is declared but its value is never read."
		},
		{
			"severity": 4,
			"source": "typescript",
			"range": {
				"end": {
					"character": 38,
					"line": 2
				},
				"start": {
					"character": 0,
					"line": 2
				}
			},
			"code": 6133,
			"message": "'utils' is declared but its value is never read."
		},
		{
			"severity": 4,
			"source": "typescript",
			"range": {
				"end": {
					"character": 8,
					"line": 6
				},
				"start": {
					"character": 6,
					"line": 6
				}
			},
			"code": 6133,
			"message": "'db' is declared but its value is never read."
		},
		{
			"severity": 4,
			"source": "typescript",
			"range": {
				"end": {
					"character": 12,
					"line": 10
				},
				"start": {
					"character": 5,
					"line": 10
				}
			},
			"code": 6133,
			"message": "'browser' is declared but its value is never read."
		}
	],
	"uri": "file:///Users/user/Projects/futon-media/src/adapters/chromium.ts"
}
```

Which is why I'm saying the change I've proposed (https://github.com/sublimelsp/LSP/pull/914#discussion_r428477992) will resolve this problem. I've even personally created an override of that `diagnostics.py` file to fix this problem.

Reply:
This is already fixed on both st3 and st4 branches (that's why I got confused) but fix is not released yet. We should release more often.

Reply:
Yes is already in the master,
We should also probably close this PR.

Reply:
I’ll make a release soon-ish.

Reply:
Version 0.11.0 is pushed which should contain the desired behavior.